feat: Add timepicker component

jz5426 opened the pull request:
for #205

jz5426 committed:
custom components

Signed-off-by: Max You <max.you@outlook.com>

zvonimirfras committed:
Restore export modal tab order between opens

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Max You <max.you@outlook.com>

Akshat55 committed:
Centralize linting rules in root (#291)

* chore: migrate nx to the latest version

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: update base branch

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: centralize linting rules in root

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: update lib project name to match supported framework

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: remove overrides

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: lint fix

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: lint fix

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: disable caching for lint results

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: update lint fix script

Signed-off-by: Akshat Patel <akshat@live.ca>

---------

Signed-off-by: Akshat Patel <akshat@live.ca>
Signed-off-by: Max You <max.you@outlook.com>

zvonimirfras committed:
Support {{children}} in custom components

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Max You <max.you@outlook.com>

zvonimirfras committed:
Fix export template not working when items empty

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Max You <max.you@outlook.com>

Akshat55 committed:
fix: add workflow to release package (#292)

* chore: update build directory names and path

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: lint fix

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: update project source root

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: Set output path for libs in dist/libs/*

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: Add nx release script

Signed-off-by: Akshat Patel <akshat@live.ca>

* ci: Add publish ci flow

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: Add SemVer versioning

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: add commintlint

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: add root workspace

Signed-off-by: Akshat Patel <akshat@live.ca>

* Support {{children}} in custom components

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Akshat Patel <akshat@live.ca>

* Fix export template not working when items empty

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: publish without packing

Signed-off-by: Akshat Patel <akshat@live.ca>

* revert: file deletion

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: give execute permissions

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: use carbon bot for publishing github releases

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: skip app commit types

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: add pp as an acceptable commit type

Signed-off-by: Akshat Patel <akshat@live.ca>

* chore: update secret name

Signed-off-by: Akshat Patel <akshat@live.ca>

* ci: migrate back to workflow

Signed-off-by: Akshat Patel <akshat@live.ca>

---------

Signed-off-by: Akshat Patel <akshat@live.ca>
Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Co-authored-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Max You <max.you@outlook.com>

Akshat55 committed:
fix: remove nx set sha action (#293)

Signed-off-by: Akshat Patel <akshat@live.ca>
Signed-off-by: Max You <max.you@outlook.com>

Akshat55 committed:
fix: add release config (#294)

Signed-off-by: Akshat Patel <akshat@live.ca>
Signed-off-by: Max You <max.you@outlook.com>

Akshat55 committed:
fix: add permission for id-token (#295)

Signed-off-by: Akshat Patel <akshat@live.ca>
Signed-off-by: Max You <max.you@outlook.com>

Akshat55 committed:
fix: add repository url to packages (#296)

Signed-off-by: Akshat Patel <akshat@live.ca>
Signed-off-by: Max You <max.you@outlook.com>

zvonimirfras committed:
Add instructions to set up a new project w/ player

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Max You <max.you@outlook.com>

zvonimirfras committed:
Fix export error when custom collections undefined

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Max You <max.you@outlook.com>

zvonimirfras committed:
Fix export error when custom collections undefined

Signed-off-by: Zvonimir Fras <zvonimir@zvonimirfras.com>
Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
draft

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
draft

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
svg timepicker

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
timpicker

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
conflict

Signed-off-by: Max You <max.you@outlook.com>

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,360 @@
+		rejectDrop={true}
+		{...rest}>
+			{
+				<TimePicker
+				disabled={componentObj.disabled}
+				invalid={componentObj.invalid}
+				invalidText={componentObj.invalidText}
+				placeholder={componentObj.placeholder}
+				hideLabel={componentObj.hideLabel}
+				labelText={componentObj.label}
+				light={componentObj.light}
+				size={componentObj.size}>
+					<TimePickerSelect disabled={componentObj.disabled} labelText="time-picker-1">
+						<SelectItem value="AM" text="AM" />
+						<SelectItem value="PM" text="PM" />
+					</TimePickerSelect>
+					<TimePickerSelect disabled={componentObj.disabled} labelText="time-picker-2">
+						{
+							componentObj.items.map((step: any, index: number) => <SelectItem
+								value={step.value}
+								text={step.text}
+								key={index}
+							/>)
+						}
+					</TimePickerSelect>
+				</TimePicker>
```
User should not be able to interact (Text input/select) with the component in the fragment pane.

Recommend passing in class that removes pointer event for input & select.

Akshat55 replied:
Also missing support for custom styling.

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,360 @@
+                    return `<ibm-timepicker
+                        ${angularClassNamesFromComponentObj(json)}
+                        label="${json.label}"
+                        ${json.light ? '[theme]="light"' : '[theme]="dark"'}
```
You're binding the variables `light` & `dark` here instead of the value. Might be worth creating a new input for theme?

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,360 @@
+                        ${angularClassNamesFromComponentObj(json)}
+                        label="${json.label}"
+                        ${json.light ? '[theme]="light"' : '[theme]="dark"'}
+                        [invalid]="${json.invalid}"
+                        [placeholder]="'${json.placeholder}'"
+                        [size]="'${json.size}'"
+                        [hideLabel]="${json.hideLabel}"
+                        [invalidText]="'${json.invalidText}'"
```
Once these are set, users will not be able to update them. These need inputs. Also, you're binding them to variables, not values.

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,360 @@
+                        (valueChange)="${nameStringToVariableString(json.codeContext?.name)}ValueChange.emit($event.value)"
+                        [value]="${nameStringToVariableString(json.codeContext?.name)}Value"
+                        [disabled]="${json.disabled}">
+                            <ibm-timepicker-select ${json.light
+                                ? '[theme]="light"'
+                                : '[theme]="dark"'
+                            }
+                            [disabled]="${json.disabled}"
```
Same as above.

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,360 @@
+                                <option selected value="AM">AM</option>
+                                <option value="PM">PM</option>
+                            </ibm-timepicker-select>
+                            <ibm-timepicker-select ${json.light
+                                ? '[theme]="light"'
+                                : '[theme]="dark"'
+                            }
+                            [disabled]="${json.disabled}"
```
Same as above.

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,360 @@
+                outputs: ({ json }) => `@Output() ${nameStringToVariableString(json.codeContext?.name)}ValueChange = new EventEmitter();`,
+                imports: ['TimePickerModule', 'TimePickerSelectModule'],
+                code: ({ json }) => {
+                    return `<ibm-timepicker
+                        ${angularClassNamesFromComponentObj(json)}
+                        label="${json.label}"
+                        ${json.light ? '[theme]="light"' : '[theme]="dark"'}
+                        [invalid]="${json.invalid}"
+                        [placeholder]="'${json.placeholder}'"
+                        [size]="'${json.size}'"
+                        [hideLabel]="${json.hideLabel}"
+                        [invalidText]="'${json.invalidText}'"
+                        (valueChange)="${nameStringToVariableString(json.codeContext?.name)}ValueChange.emit($event.value)"
+                        [value]="${nameStringToVariableString(json.codeContext?.name)}Value"
+                        [disabled]="${json.disabled}">
+                            <ibm-timepicker-select ${json.light
+                                ? '[theme]="light"'
+                                : '[theme]="dark"'
+                            }
+                            [disabled]="${json.disabled}"
+                            display="inline">
+                                <option selected value="AM">AM</option>
+                                <option value="PM">PM</option>
+                            </ibm-timepicker-select>
+                            <ibm-timepicker-select ${json.light
+                                ? '[theme]="light"'
+                                : '[theme]="dark"'
+                            }
+                            [disabled]="${json.disabled}"
+                            display="inline">
+                                ${json.items.map((step: any) => (`<option
+                                    value="${step.value}"
+                                    text="${step.text}">
+                                        ${step.text}
+                                    </option>`
+                                )).join('\n')}
+                            </ibm-timepicker-select>
+                        </ibm-timepicker>`;
+                }                                
+            }
```
See comments made above

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,360 @@
+                        </ibm-timepicker>`;
+                }                                
+            }
+		},
+		react: {
```
This is indented.

I recommend you run the linter... quite a few cases where you're using double quotes instead of single... Linter should automatically fix.

Akshat55 commented on player/react/src/lib/components/ui-time-picker.tsx:
```diff
@@ -0,0 +1,55 @@
+		// eslint-disable-next-line react/jsx-no-useless-fragment
+		return <></>;
+	}
+	return <TimePicker
+		id="time-picker"
+		light={state.light}
+		disabled={state.disabled}
+		invalid={state.invalid}
+		invalidText={state.invalidText}
+		placeholder={state.placeholder}
+		size={state.size}
+		labelText={state.label}>
+		<TimePickerSelect labelText="time-picker-1" id="time-picker-select-1">
+			<SelectItem value="AM" text="AM" />
+			<SelectItem value="PM" text="PM" />
+		</TimePickerSelect>
+		<TimePickerSelect labelText="time-picker-2" id="time-picker-select-2" >
+			{
+				state.items.map((step: any, index: number) => <SelectItem
+					value={step.value}
+					text={step.text}
+					key={index}
+				/>)
+			}
+		</TimePickerSelect>
+	</TimePicker>;
```
Should there be an event for when a value is changed?

jz5426 committed:
progress

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
double quote to singlel quote

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
double quote to singlel quote

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
slots

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
build error

Signed-off-by: Max You <max.you@outlook.com>

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+		angular: {
+			latest: {
+                inputs: ({ json }) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Value = "${json.value};
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Theme = "${json.light ? 'light' : 'dark'}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsInvalid = "${json.invalid}";
```
Should this be a string? `invalid` should be a boolean.

```suggestion
					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsInvalid = ${json.invalid};
```

Akshat55 replied:
Indentation is off for entire input value. 2nd line should not be indented AT ALL. Please see how it's done in other files.

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsInvalid = "${json.invalid}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Placeholder = "${json.placeholder}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Size = "${json.size}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsHideLabel = "${json.hideLabel}";
```
```suggestion
					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsHideLabel = ${json.hideLabel};
```

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Size = "${json.size}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsHideLabel = "${json.hideLabel}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}InvalidText = "${json.invalidText}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsDisabled = "${json.disabled}";
```
```suggestion
					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsDisabled = ${json.disabled};
```

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                                <SelectItem value='AM' text='AM' />
+                                <SelectItem value='PM' text='PM' />
+                            </TimePickerSelect>
+                            <TimePickerSelect labelText='time-picker-2' id='time-picker-select-2'>
```
Id

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                        placeholder="${json.placeholder}"
+                        hideLabel={${json.hideLabel}}
+                        light={${json.light}}>
+                            <TimePickerSelect labelText='time-picker-1' id='time-picker-select-1'>
```
Id

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                                <SelectItem value='AM' text='AM' />
+                                <SelectItem value='PM' text='PM' />
+                            </TimePickerSelect>
+                            <TimePickerSelect labelText='time-picker-2' id='time-picker-select-2'>
```
Id

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                code: ({ json }) => {
+                    return `<TimePicker
+                        ${reactClassNamesFromComponentObj(json)}
+                        id="time-picker"
```
Id

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                code: ({ json }) => {
+                    return `<TimePicker
+                        ${reactClassNamesFromComponentObj(json)}
+                        id="time-picker"
```
What if they have multiple timer pickers in a single fragment? You're hard coding an `id`, this is not best practice and would cause issues.

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                        placeholder="${json.placeholder}"
+                        hideLabel={${json.hideLabel}}
+                        light={${json.light}}>
+                            <TimePickerSelect labelText='time-picker-1' id='time-picker-select-1'>
```
Id

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                code: ({ json }) => {
+                    return `<ibm-timepicker
+                        ${angularClassNamesFromComponentObj(json)}
+                        label="${json.label}"
```
What if they want to change the label? 😅 Shouldn't label be an input as well?

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                    return `<ibm-timepicker
+                        ${angularClassNamesFromComponentObj(json)}
+                        label="${json.label}"
+						[theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
```
```suggestion
			[theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
```

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                code: ({ json }) => {
+                    return `<ibm-timepicker
+                        ${angularClassNamesFromComponentObj(json)}
+                        label="${json.label}"
+						[theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
```
Same two issues as above.

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                            <ibm-timepicker-select
+							[theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
+                            [disabled]="${nameStringToVariableString(json.codeContext?.name)}IsDisabled"
+                            display='inline'>
```
What is display? Why is it set to inline?

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,379 @@
+                                <option selected value='AM'>AM</option>
+                                <option value='PM'>PM</option>
+                            </ibm-timepicker-select>
+                            <ibm-timepicker-select 
+							[theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
+                            [disabled]="${nameStringToVariableString(json.codeContext?.name)}IsDisabled"
+                            display='inline'>
```
Identation

jz5426 committed:
delete custom compo

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
Update sdk/react/src/lib/fragment-components/a-time-picker.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

jz5426 committed:
Update sdk/react/src/lib/fragment-components/a-time-picker.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

jz5426 committed:
fix

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
Merge branch 'timepicker' of https://github.com/maxxyouu/carbon-ui-builder into timepicker

jz5426 committed:
Update sdk/react/src/lib/fragment-components/a-time-picker.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

jz5426 committed:
fix

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
Merge branch 'timepicker' of https://github.com/maxxyouu/carbon-ui-builder into timepicker

jz5426 committed:
label

Signed-off-by: Max You <max.you@outlook.com>

jz5426 committed:
s[ace

Signed-off-by: Max You <max.you@outlook.com>

Akshat55 committed:
Merge branch 'main' into timepicker

Commit:
export fix

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Akshat55 commented on player/react/src/lib/components/ui-time-picker.tsx:
```diff
@@ -0,0 +1,107 @@
+		onChange={(event: any) => {
+			sendSignal(state.id, 'valueChange', [event.value], { ...state, value: event.value });
+		}}>
+		<TimePickerSelect labelText='time-picker-1' id='time-picker-select-1'>
```
hard coding id and label text

Akshat55 commented on player/react/src/lib/components/ui-time-picker.tsx:
```diff
@@ -0,0 +1,107 @@
+			<SelectItem value='AM' text='AM' />
+			<SelectItem value='PM' text='PM' />
+		</TimePickerSelect>
+		<TimePickerSelect labelText='time-picker-2' id='time-picker-select-2' >
```
hard coding id and label text

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,376 @@
+				labelText={componentObj.label}
+				light={componentObj.light}
+				size={componentObj.size}>
+					<TimePickerSelect disabled={componentObj.disabled} labelText='time-picker-1'>
```
Hard coding label Text... Do you even need label text for this?

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,376 @@
+						<SelectItem value='AM' text='AM' />
+						<SelectItem value='PM' text='PM' />
+					</TimePickerSelect>
+					<TimePickerSelect disabled={componentObj.disabled} labelText='time-picker-2'>
```
Same as above

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,376 @@
+	codeExport: {
+		angular: {
+			latest: {
+                inputs: ({ json }) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Value = "${json.value}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Theme = "${json.light ? 'light' : 'dark'}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsInvalid = ${json.invalid};
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Placeholder = "${json.placeholder}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Size = "${json.size}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsHideLabel = ${json.hideLabel};
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}InvalidText = "${json.invalidText}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsDisabled = ${json.disabled};
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Label = "${json.label}";
+					`,
```
```suggestion
                inputs: ({ json }) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Value = "${json.value}";
                	@Input() ${nameStringToVariableString(json.codeContext?.name)}Theme = "${json.light ? 'light' : 'dark'}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}IsInvalid = ${json.invalid};
			@Input() ${nameStringToVariableString(json.codeContext?.name)}Placeholder = "${json.placeholder}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}Size = "${json.size}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}IsHideLabel = ${json.hideLabel};
			@Input() ${nameStringToVariableString(json.codeContext?.name)}InvalidText = "${json.invalidText}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}IsDisabled = ${json.disabled};
			@Input() ${nameStringToVariableString(json.codeContext?.name)}Label = "${json.label}";`,
```

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,376 @@
+                        [value]="${nameStringToVariableString(json.codeContext?.name)}Value"
+                        [disabled]="${json.disabled}">
+                            <ibm-timepicker-select
+                            [theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
+                            [disabled]="${nameStringToVariableString(json.codeContext?.name)}IsDisabled"
+                            display='inline'>
```
```suggestion
                           	 [theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
                           	 [disabled]="${nameStringToVariableString(json.codeContext?.name)}IsDisabled"
                            	display='inline'>
```

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,376 @@
+                                <option value='PM'>PM</option>
+                            </ibm-timepicker-select>
+                            <ibm-timepicker-select 
+                            [theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
+                            [disabled]="${nameStringToVariableString(json.codeContext?.name)}IsDisabled"
+                            display='inline'>
```
```suggestion
                           	 [theme]="${nameStringToVariableString(json.codeContext?.name)}Theme"
                           	 [disabled]="${nameStringToVariableString(json.codeContext?.name)}IsDisabled"
                           	 display='inline'>
```

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,376 @@
+                }
+            },
+            v10: {
+                inputs: ({ json }) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Value = "${json.value};
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Theme = "${json.light ? 'light' : 'dark'}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsInvalid = "${json.invalid}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Placeholder = "${json.placeholder}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}Size = "${json.size}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsHideLabel = "${json.hideLabel}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}InvalidText = "${json.invalidText}";
+					@Input() ${nameStringToVariableString(json.codeContext?.name)}IsDisabled = "${json.disabled}";
+					`,
```
```suggestion
                inputs: ({ json }) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Value = "${json.value};
                	@Input() ${nameStringToVariableString(json.codeContext?.name)}Theme = "${json.light ? 'light' : 'dark'}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}IsInvalid = "${json.invalid}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}Placeholder = "${json.placeholder}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}Size = "${json.size}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}IsHideLabel = "${json.hideLabel}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}InvalidText = "${json.invalidText}";
			@Input() ${nameStringToVariableString(json.codeContext?.name)}IsDisabled = "${json.disabled}";`,
```

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,376 @@
+                        placeholder="${json.placeholder}"
+                        hideLabel={${json.hideLabel}}
+                        light={${json.light}}>
+                            <TimePickerSelect labelText='time-picker-1'>
```
What is labelText? Please double check all instances of this.

jz5426 committed:
Update sdk/react/src/lib/fragment-components/a-time-picker.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

jz5426 committed:
Update sdk/react/src/lib/fragment-components/a-time-picker.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

jz5426 committed:
Update sdk/react/src/lib/fragment-components/a-time-picker.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

jz5426 committed:
Update sdk/react/src/lib/fragment-components/a-time-picker.tsx

Co-authored-by: Akshat Patel <38994122+Akshat55@users.noreply.github.com>

Commit:
pr revision

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

zvonimirfras reviewed (changes requested):
AM/PM and Timezone should be optional. They also shouldn't appear in the default component object.

Commit:
make am/pm and timezone optional

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Commit:
make am/pm and timezone optional

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -63,6 +63,22 @@ export const ATimePickerSettingsUI = ({ selectedComponent, setComponent }: any)
+			onChange={(_: any, { checked }: any) => setComponent({
+				...selectedComponent,
+				timezone: checked
+		})} />
```
Instead can't we just conditionally render based on timezone length? If items length is 0, we don't render

Commit:
done

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Commit:
fix export

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

maxxyouu commented:
Verified that the component features and export works.

Commit:
fix action

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,396 @@
+									<SelectItem value='PM' text='PM' />
+								</TimePickerSelect>` : ``
+							}
+							${json.timezone ? 
```
```suggestion
							${json.items.length > 0 ?
```

Akshat55 commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,396 @@
+									<SelectItem value='PM' text='PM' />
+								</TimePickerSelect>` : ``
+							}
+							${json.timezone ? 
```
```suggestion
							${json.items.length > 0 ?
```

Akshat55 reviewed (changes requested):
Time picker timezones do not show up for react because `timezone` does not exist.

zvonimirfras commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,396 @@
+				invalid: checked
+		})} />
+		<Dropdown
+			id="size"
```
```suggestion
			id='size'
```

Commit:
Merge branch 'main' of https://github.com/IBM/carbon-ui-builder into timepicker

Commit:
lint

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Akshat55 committed:
chore: lint fix

fix: conditionally render time picker select in export and give user control to render
Signed-off-by: Akshat Patel <akshat@live.ca>

Akshat55 committed:
chore: lint exports

Signed-off-by: Akshat Patel <akshat@live.ca>

Akshat55 committed:
fix: remove redundant attribute from interface

Akshat55 committed:
fix: correct export conditions

Signed-off-by: Akshat Patel <akshat@live.ca>

Akshat55 committed:
fix: add missing events in export

Signed-off-by: Akshat Patel <akshat@live.ca>

Commit:
fixed angular export

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Commit:
angular export

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Commit:
resolve conflict

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Commit:
angular export fix

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Commit:
fix react export import section

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Akshat55 committed:
Merge branch 'main' into timepicker

jz5426 committed:
Update player/react/src/lib/components/ui-time-picker.tsx

Co-authored-by: Zvonimir Fras <zvonimir.fras@gmail.com>

jz5426 committed:
Update player/react/src/lib/components/ui-time-picker.tsx

Co-authored-by: Zvonimir Fras <zvonimir.fras@gmail.com>

jz5426 committed:
Update sdk/react/src/lib/fragment-components/a-time-picker.tsx

Co-authored-by: Zvonimir Fras <zvonimir.fras@gmail.com>

Commit:
revision

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Commit:
revision

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

maxxyouu commented on player/react/src/lib/components/ui-time-picker.tsx:
```diff
@@ -0,0 +1,119 @@
+	...commonSlots,
+	...slotsDisabled,
+	invalid: 'boolean',
+	isInvalid: (state: TimePickerState) => ({
```
i think the naming of the setter is fine, i don't think other component (i,e. combobox) use this naming. `setInvalid` is fine but no need for unnecessary changes. 

In terms of the need of `setInvalid`, please check combobox as well because the setter is similar.

maxxyouu commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,487 @@
+				selected: false
+			}
+		],
+		value: '',
```
what is wrong with empty value?

maxxyouu commented on sdk/react/src/lib/fragment-components/a-time-picker.tsx:
```diff
@@ -0,0 +1,487 @@
+			}
+		],
+		value: '',
+		timezones: []
```
there is no default timezone values so i make it empty initially

Commit:
revision

Signed-off-by: Max You <JianzhongMax.You@ibm.com>

Commit:
style

Signed-off-by: Max You <JianzhongMax.You@ibm.com>
